test/performance/observed_concurency_test.go

```diff
@@ -51,16 +56,84 @@ func createTestCase(val float32, name string) testgrid.TestCase {
 	return tc
 }
 
-func waitForServiceLatestCreatedRevision(clients *test.Clients, names test.ResourceNames) (string, error) {
-	var revisionName string
-	err := test.WaitForServiceState(clients.ServingClient, names.Service, func(s *v1alpha1.Service) (bool, error) {
-		if s.Status.LatestCreatedRevisionName != names.Revision {
-			revisionName = s.Status.LatestCreatedRevisionName
-			return true, nil
+// generateTraffic loads the given endpoint with the given concurrency for the given duration.
+// All responses are forwarded to a channel, if given.
+func generateTraffic(client *spoof.SpoofingClient, url string, concurrency int, duration time.Duration, resChannel chan *spoof.Response) (int32, error) {
```

Comment:
do you think we need a common non-fortio way of sending load for more than this test? Maybe this method should be in performance.go?

Reply:
In fact, this method is written in a very generalized way, yes. I kinda wanted to keep it separate until a proper need arises to not confuse people to chose between either.

If you feel strongly about it, I can move it immediately though.

Reply:
I understand that the main difference with our fortio wrapper is the ability to process the responses. Are we sure there's no way to expose that in fortio?

Reply:
@alculquicondor I'm fairly certain (read through the source code). Only the length of the responses will be reported.

Reply:
yeah. I did not find a way to parse the response object either. Maybe this is something we can check with the team.

Reply:
I agree we can address that upstream. I feel like if that shouldn't block this PR though, what's your take on that?

Reply:
yes. we can discuss this offline. not blocking the PR on that.